### PR TITLE
Minor improvements to the test failure messages.

### PR DESCRIPTION
**Pull request by vivekmenezes**

### CODE REVIEW

File: sql/logic_test.go [hunk 1]
```diff
@@ -376,10 +376,14 @@ func (t *logicTest) execStatement(stmt logicStatement) {
 	switch {
 	case stmt.expectErr == "":
 		if err != nil {
-			t.Fatalf("%s: expected success, but found %v", stmt.pos, err)
+			t.Fatalf("%s: expected success, but found %q", stmt.pos, err)
```
Comment (tschottdorf) [inline sql/logic_test.go]: printing `error` as `%q` is new to me. Do you have an explanation?

Comment (vivekmenezes) [inline sql/logic_test.go]: In this case I've copied it from below where error was compared to a user
supplied string. I think using %q is valid for that case and I've changed
this for consistency.

Comment (tschottdorf) [inline sql/logic_test.go]: ah, here's the corresponding piece in the docs:

>
If the format (which is implicitly %v for Println etc.) is valid for a string (%s %q %v %x %X), the following two rules apply:
>
3. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).

I didn't know about this special casing.

Comment (MBerhault) [inline sql/logic_test.go]: I changed some of them earlier to make be able to tell when the problem was due to escaping. much more convenient that way.

**Comment by MBerhault** [comment]
LGTM

**Pull request merged**